### ISSUE TITLE
Generic type with notnull constraint complains about possible null reference when using default

### ISSUE THREAD

**Issue by krwq** [dotnet/roslyn#46412]
**Version Used**: master (26 Jul 2020) 

**Steps to Reproduce**:

```csharp
#nullable enable
public class C<V> where V : notnull
{
    public void Test()
    {
        M(default(V)); // <-- warning
    }
    
    public void M(V value)
    {
    }
}
```

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8DEA7AVwwwmAzgAI49TyBYAKAAEBmCpgJgoGEAeANQB8FAO4ALRJX4UQFPAHsYhYowDejCpvZsmAFgoAVOLAAUASg1b1DLbYoBZEwBM4AMwhEYJ/mbMBuS00AX0CKUNZ2fUdpADcIDAI4CxsrUJCGIKA===

**Expected Behavior**:

No warnings

**Actual Behavior**:

```
warning CS8604: Possible null reference argument for parameter 'value' in 'void C<V>.M(V value)'.
```

cc: @dotnet/nullablefc

**Comment by safern** [comment]
I think this behavior is correct since default for a reference type returns null.


https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAgrgOyQExAagB8BiTdAG3ImHLgAI5NraBYAKAAEBmOjgJjoBhdgG92dCbz59xksW0mSAYgHsVACgCUAblkSAvuz28eHAIwA2XgBY6qjZuPzFE8wAYA/HQAedALx0OHAAZhAUMLoKLuYAdOYAnOoAJABEIt5+fmSU+ik6xoZs+kA==

**Comment by stephentoub** [comment]
> Expected Behavior: No warnings

Why?

**Comment by krwq** [comment]
@stephentoub I thought `notnull` means value type minus `Nullable<T>`... If that's not correct then I'll close the issue

**Comment by krwq** [comment]
Ok my bad 🤦

**Comment by krwq** [comment]
I think getting a default value of something which can't be null should not return null or should produce error instead of warning since it just doesn't make sense
